### PR TITLE
Enhance `ctc_loss` to return forward probs.

### PR DESCRIPTION
**Pull request by yotarok**
Continuing from #293, we found some use cases where interim alpha probabilities are useful.
This PR renames `ctc_loss` to `ctc_loss_with_foward_probs` and make `ctc_loss` to be an alias to `ctc_loss_with_forward_probs` with omitting auxiliary outputs.

### CODE REVIEW

File: optax/_src/loss.py [hunk 1]
```diff
@@ -352,9 +371,28 @@ def loop_body(prev, x):
...
+  """Computes CTC loss.
+
+  See docstring for `ctc_loss_with_forward_score` details.
+
+  Returns:
+    (B,)-array containing loss values for each sequence in the batch.
+  """
+  return ctc_loss_with_forward_probs(
+      logits, logitpaddings, labels, labelpaddings,
+      blank_id=blank_id, logepsilon=logepsilon)[0]
```
Comment (hbq1) [inline optax/_src/loss.py:398]: Could you change it to
```
per_seq_loss, _, _ = ctc_loss_with_forward_probs(...)
return per_seq_loss
```
for better readability? (indexing is easy to overlook)

**Review by hbq1 — APPROVED** [review]
Thanks a lot for the detailed comments! LGTM

File: optax/_src/loss.py (hunk 1 above)
Comment (yotarok) [reply optax/_src/loss.py]: Thanks for the review. Done.

**Comment by yotarok** [comment]
Copied args section of docstring for `ctc_loss_with_forward_probs` to `ctc_loss`.

**Comment by yotarok** [comment]
I just modified docstring for making it more friendly with internal lint service, and now it's done.
